src/olympia/addons/models.py

```diff
@@ -357,13 +360,35 @@ def get_queryset_for_pending_queues(
                     & ~Q(**{listed_delay_flag_field: datetime.max})
                 )
             )
+        needs_human_review_kw = {
```

Comment:
I can't easily submit this as a suggestion because it touches lines outside of your patch, but what do you think of this:

```diff
diff --git a/src/olympia/addons/models.py b/src/olympia/addons/models.py
index 0667e5fd37..4524fe57df 100644
--- a/src/olympia/addons/models.py
+++ b/src/olympia/addons/models.py
@@ -321,7 +321,7 @@ class AddonManager(ManagerBase):
             select_related_fields_for_listed=False,
         )
         versions_due_qs = (
-            Version.unfiltered.filter(due_date__isnull=False)
+            Version.unfiltered.filter(due_date__isnull=False, addon=OuterRef('pk'))
             .no_transforms()
             .order_by('due_date')
         )
@@ -360,32 +360,26 @@ class AddonManager(ManagerBase):
                     & ~Q(**{listed_delay_flag_field: datetime.max})
                 )
             )
-        needs_human_review_kw = {
-            'addon_id': OuterRef('pk'),
-            'needshumanreview__is_active': True,
-        }
         qs = (
             qs.filter(**filters)
             .annotate(
                 first_version_due_date=Min('versions__due_date'),
-                first_version_id=Subquery(
-                    versions_due_qs.filter(addon=OuterRef('pk')).values('pk')[:1]
-                ),
+                first_version_id=Subquery(versions_due_qs.values('pk')[:1]),
                 needs_human_review_from_cinder=Exists(
                     versions_due_qs.filter(
-                        **needs_human_review_kw,
+                        needshumanreview__is_active=True,
                         needshumanreview__reason=NeedsHumanReview.REASONS.CINDER_ESCALATION,
                     )
                 ),
                 needs_human_review_from_abuse=Exists(
                     versions_due_qs.filter(
-                        **needs_human_review_kw,
+                        needshumanreview__is_active=True,
                         needshumanreview__reason=NeedsHumanReview.REASONS.ABUSE_ADDON_VIOLATION,
                     )
                 ),
                 needs_human_review_from_appeal=Exists(
                     versions_due_qs.filter(
-                        **needs_human_review_kw,
+                        needshumanreview__is_active=True,
                         needshumanreview__reason=NeedsHumanReview.REASONS.ABUSE_ADDON_VIOLATION_APPEAL,
                     )
                 ),

```

It avoids repeating the `OuterRef` and removes the need for the weird dict (unfortunately we can't simplify this further and build a `needs_human_review_qs` ahead of time since the `reason` need to live in the same `filter()` to be in the same `JOIN`).

Reply:
>  we can't simplify this further and build a needs_human_review_qs ahead of time since the reason need to live in the same filter() to be in the same JOIN

that answers my question as to why my attempts at optimizing by having a qs object that had all the common kwargs failed 🥲